src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs

```diff
@@ -132,7 +132,7 @@ INegatedPatternOperation or
             {
                 return ValueUsageInfo.ReadableWritableReference;
             }
-            else if (operation.Parent is IIncrementOrDecrementOperation)
```

Comment:
@mavasani This implementation has a copy in roslyn-analyzers:

https://github.com/dotnet/roslyn-analyzers/blob/main/src/Utilities/Compiler/Extensions/IOperationExtensions.cs

Can we investigate having an automation similar to dotnet/aspnetcore and dotnet/runtime (example: https://github.com/dotnet/runtime/pull/72588)

Reply:
Interesting, I think we have discussed about making some of this functionality available through public API. Not sure how acceptable that would be.

Reply:
I'm not sure if the API is accurate/reliable enough to be public. But it would be nice to have public (after any needed improvements, if any)